org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/compiler/ErrorSafeExtensions.java

```diff
@@ -0,0 +1,166 @@
+/**
+ * Copyright (c) 2012, 2020 itemis AG (http://www.itemis.eu) and others.
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.xtext.xbase.compiler;
+
+import com.google.common.base.Objects;
+import com.google.common.collect.Iterables;
+import com.google.inject.Inject;
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.xtext.common.types.JvmAnnotationReference;
+import org.eclipse.xtext.common.types.JvmSpecializedTypeReference;
+import org.eclipse.xtext.common.types.JvmTypeReference;
+import org.eclipse.xtext.common.types.JvmUnknownTypeReference;
+import org.eclipse.xtext.diagnostics.Severity;
+import org.eclipse.xtext.validation.Issue;
+import org.eclipse.xtext.xbase.compiler.BrokenTypeRefDetector;
+import org.eclipse.xtext.xbase.compiler.IElementIssueProvider;
+import org.eclipse.xtext.xbase.compiler.LoopParams;
+import org.eclipse.xtext.xbase.compiler.TypeReferenceSerializer;
+import org.eclipse.xtext.xbase.compiler.output.ErrorTreeAppendable;
+import org.eclipse.xtext.xbase.compiler.output.ITreeAppendable;
+import org.eclipse.xtext.xbase.lib.IterableExtensions;
+import org.eclipse.xtext.xbase.lib.ObjectExtensions;
+import org.eclipse.xtext.xbase.lib.Procedures.Procedure1;
+import org.eclipse.xtext.xbase.lib.Procedures.Procedure2;
+
+/**
+ * @author Jan Koehnlein
+ */
+public class ErrorSafeExtensions {
+	@Inject
+	private TypeReferenceSerializer typeReferenceSerializer;
+
+	@Inject
+	private IElementIssueProvider.Factory issueProviderFactory;
+
+	public Iterable<Issue> getErrors(EObject element) {
+		IElementIssueProvider issueProvider = issueProviderFactory.get(element.eResource());
+		return Iterables.filter(issueProvider.getIssues(element), it -> Severity.ERROR.equals(it.getSeverity()));
+	}
+
+	public boolean hasErrors(EObject element) {
+		IElementIssueProvider issueProvider = issueProviderFactory.get(element.eResource());
```

Comment:
I know it's not really the scope of this PR, but `return getErrors().iterator().hasNext()` avoid a little bit of duplication here. Or even `!Iterables.isEmpty(getErrors())`